### ISSUE TITLE
[EDITOR] Grid moving along a selected object in 2D view under certain conditions.

### ISSUE THREAD

**Issue by Lwiredthefirst** [godotengine/godot#49448]
### Godot version

3.3.2.stable.official

### System information

Windows 10 GLES 3 GTX1060

### Issue description

In 2D view. 
1/ When "View" > "Always Show Grid" option is ticked **and** "Use Grid Snap" is **not** selected, if the user enters "Select Mode" ,"Move Mode" , "Rotate Mode" or "Scale Mode" and tries to move, rotate or scale, an object on the grid, the grid moves along the object. 

2/When "View" > "Always Show Grid" option is ticked **and** "Use Grid Snap" **is**  selected, if the user enters "Rotate Mode" or "Scale Mode" and tries rotate or scale, an object on the grid, the grid moves along the object. 

The grid should never follow the object, it is very confusing because it is very confusing and it is not how it works in classic 2D editors.

### Steps to reproduce

Please see issue description.

**Comment by kleonc** [comment]
Can't reproduce unless "Snap Relative" is checked. And when "Snap Relative" is checked then I'd say grid lines should indeed croos at the position of the selected object. However, it seems that grid is being snapped to the object's rect position in the global canvas coordinate space instead:
![2sgsaMbYib](https://user-images.githubusercontent.com/9283098/121278906-de68d900-c8d3-11eb-8f70-d8c233e0c842.gif)

no idea if that's intended behavior.

**Comment by Calinou** [comment]
cc @groud

**Comment by groud** [comment]
Yeah, I agree the grid should be kept aligned with the position, not the encompassing rect.

**Comment by kleonc** [comment]
@groud Note that grid is being aligned to the rect encompassing all of the selected nodes (CanvasItems?). So for single item selected it makes sense to align grid to the position but not sure if anything other than aligning to the encompassing rect makes sense if more items are selected.
